src/main/java/org/openstreetmap/atlas/geography/sharding/DynamicTileSharding.java

```diff
@@ -448,6 +447,55 @@ protected SwitchList switches()
                 GEOJSON);
     }
 
+    /**
+     * Calculates and saves the counts for each zoom layer lower than firstZoomLayerToGenerate.
+     *
+     * @param firstZoomLayerToGenerate
+     *            the first zoom layer for which to generate counts
+     * @param countsAtHigherZoom
+     *            HashMap containing counts for all {@link SlippyTile}s in
+     *            firstZoomLayerToGenerate+1
+     * @return a HashMap containing counts for all (@link SlippyTile}s in zoomLayerToGenerate and
+     *         below
+     */
+    private HashMap<SlippyTile, Long> calculateTileCountsForAllZoom(
+            final int firstZoomLayerToGenerate, final HashMap<SlippyTile, Long> counts)
+    {
+        for (int currentZoom = firstZoomLayerToGenerate; currentZoom >= 0; currentZoom--)
+        {
+            long count = 0;
+            long tilesCalculated = 0;
+            for (int x = 0; x < Math.pow(2, currentZoom + 1); x += 2)
+            {
+                for (int y = 0; y < Math.pow(2, currentZoom + 1); y += 2)
```

Comment:
Just a thought: `Math.pow()` can often be a pretty slow function, since it must work for the general case (doubles) using log-based formulas under the hood. In specific cases it can be worth it to roll a custom `pow()` function if you know you're only operating on integers. That being said it's not likely to be a performance bottleneck here, since the outer zoom loop is not particularly tight.

As another note: hopefully the JIT compiler is optimizing those `pow()` calls out of the inner loop conditions so that it is not called on every iteration (considering you've got nested loops here). I would imagine it's smart enough to do this, so this note is probably moot.